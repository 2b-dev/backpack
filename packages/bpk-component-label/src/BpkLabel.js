import PropTypes from 'prop-types';
import React from 'react';

import './bpk-label.scss';

const BpkLabel = (props) => {
  const { children, required, white, disabled, className, ...rest } = props;
  const classNames = ['bpk-label'];

  if (white) { classNames.push('bpk-label--white'); }
  if (disabled) { classNames.push('bpk-label--disabled'); }
  if (className) { classNames.push(className); }

  return (
    // The BpkFieldset component addresses the issue of enforcing `htmlFor`
    // eslint-disable-next-line jsx-a11y/label-has-for
    <label className={classNames.join(' ')} {...rest}>
      {children}
      {required && (
        <span className="bpk-label__asterix">*</span>
      )}
    </label>
  );
};

BpkLabel.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  white: PropTypes.bool,
};

BpkLabel.defaultProps = {
  className: null,
  disabled: false,
  required: false,
  white: false,
};

export default BpkLabel;
