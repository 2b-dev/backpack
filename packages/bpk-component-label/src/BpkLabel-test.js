import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import BpkLabel from './BpkLabel'

describe('BpkLabel', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkLabel label='Origin' htmlFor='origin' />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with a "white" attribute', () => {
    const tree = renderer.create(<BpkLabel label='Origin' htmlFor='origin' white />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

