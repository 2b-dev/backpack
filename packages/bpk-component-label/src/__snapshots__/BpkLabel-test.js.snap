exports[`BpkLabel should render correctly 1`] = `
<label
  className="bpk-label"
  htmlFor="origin">
  Origin
</label>
`;

exports[`BpkLabel should render correctly with a "white" attribute 1`] = `
<label
  className="bpk-label bpk-label--white"
  htmlFor="origin">
  Origin
</label>
`;
